### PR TITLE
FISH-13392: enabling unit tests after fixing issue for NullPointerException

### PR DESCRIPTION
**Pull request by breakponchito**
Enabling unit test for Fault Tolerance previously disabled
## Description

This is a fix to enable the unit test that previously were disabled when upgrading the version of Fault Tolerance for MicroProfile 7.1
### Testing Performed

Build server from pipeline without any issue after passing all unit tests and the TCK from Fault Tolerance:

image

Fault Tolerance TCK:
image

### Testing Environment

Ubuntu 20.04, azul jdk 21,  maven 3.9.11

### CODE REVIEW

File: appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java [hunk 1]
```diff
@@ -344,38 +349,54 @@ default FaultToleranceMetrics boundTo(FaultToleranceMethodContext context, Fault
...
         return this;
     }
+
+    default Meter getCurrentMeter() {
+        if (getDefaultHabitat() == null) {
+            return null;
+        }
+        OpenTelemetryService service = getDefaultHabitat().getService(OpenTelemetryService.class);
+        return service != null ? service.getCurrentMeter() : null;
+    }
```
Comment (luiseufrasio) [inline appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java:402]: Would it be better reusing the call to `getDefaultHabitat`:
```
default Meter getCurrentMeter() {
      var habitat = getDefaultHabitat();
      if (habitat == null) return null;
      var service = habitat.getService(OpenTelemetryService.class);
      return service != null ? service.getCurrentMeter() : null;
  }
```

WDYT?

File: appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java [hunk 2]
```diff
@@ -302,17 +302,18 @@ public String getClassAndMethodName() {
...
                     {"result", "valueReturned", "exceptionThrown"}, fallbackTag});
             //place to register telemetry ft.invocations.total
-            Meter currentMeter = openTelemetryService.getCurrentMeter();
-            FaultToleranceMethodContextImpl faultToleranceMethodContext = (FaultToleranceMethodContextImpl) context;
-            setClassAndMethodName(faultToleranceMethodContext.getClassName() + "." + faultToleranceMethodContext.getMethodName());
-            addFTInvocationTotalMeter(createFTInvocationTotalMeter(getClassAndMethodName(), currentMeter, policy.isFallbackPresent()));
+            Meter currentMeter = getCurrentMeter();
+            if (currentMeter != null) {
+                FaultToleranceMethodContextImpl faultToleranceMethodContext = (FaultToleranceMethodContextImpl) context;
+                setClassAndMethodName(faultToleranceMethodContext.getClassName() + "." + faultToleranceMethodContext.getMethodName());
```
Comment (luiseufrasio) [inline appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java]: Should `setClassAndMethodName` be done before `if (currentMeter != null)` instead of inside it?

File: appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java [hunk 3]
```diff
@@ -344,38 +349,54 @@ default FaultToleranceMetrics boundTo(FaultToleranceMethodContext context, Fault
...
+                        addFTBulkheadWaitingDuration(createFTBulkheadWaitingDuration(currentMeter));
+                    }
                 } else {
                     AtomicInteger running = context.getQueuingOrRunningPopulation();
                     register("ft.bulkhead.executionsRunning", null, running::get);
                     this.setExecutionBulkheadRunningSupplier(running::get);
-                    addFTBulkheadExecutionRunning(createFTBulkheadExecutionsRunning(currentMeter, this));
+                    if (currentMeter != null) {
+                        addFTBulkheadExecutionRunning(createFTBulkheadExecutionsRunning(currentMeter, this));
+                    }
```
Comment (luiseufrasio) [inline appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java:387]: Is there a way to avoid the repetition of `if (currentMeter != null)` ?

**Review by luiseufrasio — CHANGES_REQUESTED** [review]

File: appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java (hunk 1 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java:402]: done

File: appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java (hunk 3 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java:387]: We have different conditions from the policy to validate if we can apply specific configuration, in this case it is needed to validate if the currentMeter is null on each condition and also to pass the junit test execution skipping null reference that can be provided by the lookup of the service

File: appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java (hunk 2 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/FaultToleranceMetrics.java]: done

**Comment by breakponchito** [comment]
Jenkins test plese

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Review by luiseufrasio — APPROVED** [review]
LGTM
